library/src/main/java/com/novoda/downloadmanager/DownloadMigrationService.java

```diff
@@ -0,0 +1,27 @@
+package com.novoda.downloadmanager;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.WorkerThread;
+
+public interface DownloadMigrationService {
+
+    MigrationFuture startMigration(NotificationChannelCreator notificationChannelCreator, NotificationCreator<MigrationStatus> notificationCreator);
+
+    interface MigrationFuture {
```

Comment:
You can remove this interface if you just pass the callback in the `migrate` method as it was before. Also, the `.observe()` is mandatory now because the execution now is delayed. What's the advantage from the previous implementation?